sycl/tools/xpti_helpers/tuple_view.hpp

```diff
@@ -0,0 +1,88 @@
+//==---------- tuple_view.hpp - Tuple View ---------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/common.hpp>
+
+#include <functional>
+#include <tuple>
+#include <type_traits>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace xpti_helpers {
+/// A helper class to provide tuple-like access to a contiguous chunk of memory.
+template <typename... Ts> struct tuple_view { unsigned char *data; };
+
+template <size_t I, typename T> struct tuple_view_element;
+
+template <size_t I, typename Head, typename... Rest>
+struct tuple_view_element<I, tuple_view<Head, Rest...>>
```

Comment:
Could you please tell why `tuple_view` classes are required?

Reply:
I don't know any other name to get a tuple-like view into raw memory. Even if reinterpret cast to `std::tuple` works, this sounds like too much UB.

Reply:
Probably it's even worse, but anyway take a look at unpack2 function in  https://godbolt.org/z/eWPhdna4W

Reply:
@romanovvlad I tried casting to `std::tuple<...> *`, but apparently it causes some problems with memory alignment. So, I've settled with your second proposal.